aptos-move/mvhashmap/src/versioned_data.rs

```diff
@@ -115,14 +133,110 @@ impl<V: TransactionWrite> Default for VersionedValue<V> {
     }
 }
 
-impl<V: TransactionWrite> VersionedValue<V> {
-    fn read(&self, txn_idx: TxnIndex) -> anyhow::Result<MVDataOutput<V>, MVDataError> {
+// TODO(BlockSTMv2): remove TransactionWrite trait requirement from V, even if
+// AggregatorV1 code is not removed by definining a more specialized trait that can
+// runtime assert in other variants. Add other variants to stored cells that allow
+// storing group metadata and group size together at the group key (currently metadata
+// is stored in a fake write entry, and group size is stored in MVGroupData map).
+impl<V: TransactionWrite + PartialEq> VersionedValue<V> {
+    // Extracts read dependencies from the data-structure that are affected by a write
+    // of 'data' at 'txn_idx'. Some of these dependencies that remain valid can be
+    // relocated by a caller to a different (new) entry in the data-structure. The
+    // boolean flag indicates whether the dependency is still valid after the write.
+    fn get_affected_read_dependencies<const ONLY_COMPARE_METADATA: bool>(
```

Comment:
I think renaming this to `extract` or `split_of` is better `get_affected_read_dependencies` sounds immutable, but the implementation actually extracts dependencies aka least upper bound for all higher indices? Commenting why it is okay to process any > 0 incarnations is also helpful.